### DISCUSSION TITLE
Is there a cmake flag to set to build dynamically and not statically?

### DISCUSSION THREAD

**Discussion by nsarzyns** [OpenDDS/OpenDDS#4720] [Q&A] [answered]
Using the configure file, the generated sln produces a dynamic build, while a cmake build is static.

I've tried running cmake with the following flags added with no luck; -DOPENDDS_STATIC=OFF -DBUILD_SHARED_LIBS=OFF. They update in the config.cmake file but does seem to have an effect on the build.

Am I missing something or does the cmake build not support a dynamic build?

**Comment by iguessthislldo** [comment, marked as answer]
CMake produces static libraries and executables by default regardless of how OpenDDS was built. Use `-DBUILD_SHARED_LIBS=ON` to get it to produce dynamic/shared libraries.

**Comment by nsarzyns** [comment]
Thank you! I'm not sure how but I think somehow I convinced myself that 'BUILD_SHARED_LIBS' was 'BUILD_STATIC_LIBS'

Now if only the ACE/TAO stuff was also installed with the install routine. I see the issue you created, #4578 and understand the complexity but as stop-gap, would using install(SCRIPT...) work to fill the void? It's known where the built items lie in the ace_tao dir. During an install routine, an external script could be referenced to copy the needed files.

**Reply by iguessthislldo** [reply]
Yes, that's one way to deal with it. If you only need runtime libraries and exes, we do support [`install(IMPORTED_RUNTIME_ARTIFACTS`](https://opendds.readthedocs.io/en/master/devguide/building/cmake.html#install-imported-runtime-artifacts). If you need a development install, you might be able to combine `install(IMPORTED_RUNTIME_ARTIFACTS)` with the ACE/TAO headers and other files needed for the full development install. I think you would get at least 95% of the files you need by just installing all the `.h` and `.inl` files in `ACE/ace` and `TAO/tao`. After that as far as I know the only thing you'd be missing on Windows is the `.lib` files.

I've never tried any of this by the way, just thought about it, so I don't know if this would actually work like this. I decided awhile I didn't really want to put any more effort into CMake workarounds for ACE/TAO except for bug fixes.

There's also a bit more info in #802 about installing on Windows. #4578 should have linked to that, but I just noticed the link was broken.
